app/iot/devices.py

```diff
@@ -1,64 +1,63 @@
-import time
+import asyncio
 
-from .message import MessageType
+from app.iot.message import MessageType
 
 
 TIME_TO_SLEEP = 0.5
 
 
-# of course this code looks dumb, but imagine some real implementations of each method here
 class HueLightDevice:
```

Comment:
Code Duplication: The classes `HueLightDevice`, `SmartSpeakerDevice` and `SmartToiletDevice` have similar methods and structure. This is a clear sign that these classes could inherit from a common base class. This would reduce code duplication and make the code easier to maintain.